samples/samples/admin/backup_snippet.py

```diff
@@ -0,0 +1,556 @@
+                name="{}/backups/{}".format(instance.name, backup_id)
+            )
+        )
+    except NotFound:
+        print("Backup creation was successfully cancelled.")
+        return
+    print("Backup was created before the cancel completed.")
+    spanner_client.database_admin_api.delete_backup(
+        backup_pb.DeleteBackupRequest(
+            name="{}/backups/{}".format(instance.name, backup_id)
+        )
+    )
+    print("Backup deleted.")
+
+
+# [END spanner_cancel_backup_create]
+
+
+# [START spanner_list_backup_operations]
+def list_backup_operations(instance_id, database_id, backup_id):
+    spanner_client = spanner.Client()
+    instance = spanner_client.instance(instance_id)
+
+    # List the CreateBackup operations.
+    filter_ = (
+        "(metadata.@type:type.googleapis.com/"
+        "google.spanner.admin.database.v1.CreateBackupMetadata) "
+        "AND (metadata.database:{})"
+    ).format(database_id)
+    request = backup_pb.ListBackupOperationsRequest(
+        parent=instance.name, filter=filter_
+    )
+    operations = spanner_client.database_admin_api.list_backup_operations(request)
+    for op in operations:
+        metadata = protobuf_helpers.from_any_pb(
+            backup_pb.CreateBackupMetadata, op.metadata
+        )
+        print(
+            "Backup {} on database {}: {}% complete.".format(
+                metadata.name, metadata.database, metadata.progress.progress_percent
+            )
+        )
+
+    # List the CopyBackup operations.
+    filter_ = (
+        "(metadata.@type:type.googleapis.com/google.spanner.admin.database.v1.CopyBackupMetadata) "
+        "AND (metadata.source_backup:{})"
+    ).format(backup_id)
+    request = backup_pb.ListBackupOperationsRequest(
+        parent=instance.name, filter=filter_
+    )
+    operations = spanner_client.database_admin_api.list_backup_operations(request)
+    for op in operations:
+        metadata = protobuf_helpers.from_any_pb(
+            backup_pb.CopyBackupMetadata, op.metadata
+        )
+        print(
+            "Backup {} on source backup {}: {}% complete.".format(
+                metadata.name,
+                metadata.source_backup,
+                metadata.progress.progress_percent,
+            )
+        )
+
+
+# [END spanner_list_backup_operations]
+
+
+# [START spanner_list_database_operations]
+def list_database_operations(instance_id):
+    spanner_client = spanner.Client()
+    instance = spanner_client.instance(instance_id)
+
+    # List the progress of restore.
+    filter_ = (
+        "(metadata.@type:type.googleapis.com/"
+        "google.spanner.admin.database.v1.OptimizeRestoredDatabaseMetadata)"
+    )
+    request = spanner_database_admin.ListDatabaseOperationsRequest(
+        parent=instance.name, filter=filter_
+    )
+    operations = spanner_client.database_admin_api.list_database_operations(request)
+    for op in operations:
+        metadata = protobuf_helpers.from_any_pb(
```

Comment:
same here

Reply:
All the languages should be consistent in decoding operations for a specifc samples, we are trying to follow what is present in:

Golang= https://github.com/GoogleCloudPlatform/golang-samples/blob/main/spanner/spanner_snippets/spanner/spanner_list_database_operations.go#L58

Java=https://github.com/googleapis/java-spanner/blob/main/samples/snippets/src/main/java/com/example/spanner/SpannerSample.java#L1735-L1742